packages/http-server/src/server.ts

```diff
@@ -28,15 +28,14 @@ function optionallySerializeAndSend(
   const serializer = respSerializers.find((s: Serializer) => s.test(contentType || ''));
 
   const data = serializer && output.body ? serializer.serializer(output.body) : output.body;
+  const isInUTF8 = contentType && contentType.includes('charset=utf-8');
 
   if (serializer) {
     reply.serializer(serializer.serializer);
   }
 
-  // TODO: do we need charset=utf-8 at all ???
-  if (output && output.headers && output.headers['Content-type'].includes('application/json')) {
-    // why would only application/json have the charset?
-    reply.header('content-type', output.headers['Content-type'] + '; charset=utf-8');
+  if (contentType && contentType.includes('application/json')) {
```

Comment:
This TODO was asking a good question, why are we trying to ensure utf-8 is specified in there? Stuff might not be, what if the content type is `application/xml; charset=iso-8859-1`, would this make it: `application/xml; charset=iso-8859-1; charset=utf-8`?

@XVincentX do you know what the original intent was here?

Reply:
could be something to consider for another PR :+1:

Reply:
Sadly it is directly related to this code because you are adding this functionality in this PR and I am not sure we should be adding this functionality.

Reply:
@philsturgeon, I'm not adding anything big really. I just fixed some discrepancies. In most of the tests, json responses had `charset=utf-8`, so I added that to the responses that didn't have it.